client/src/components/ExpenseForm.tsx

```diff
@@ -0,0 +1,272 @@
+    setIsProcessing(true);
+
+    if (variant === 'new') {
+      addExpense({
+        user: formUser,
+        title: formTitle,
+        category: formCategory,
+        amount: formAmount,
+        date: formDate,
+        note: formNote,
+      })
+        .then(() => {
+          dispatch(setModalState({ variant: null }));
+          dispatch(fetchExpenses());
+        })
+        .catch(() => {
+          setIsProcessing(false);
+        });
+    }
+
+    if (variant === 'edit') {
+      if (!currentExpense) {
+        setIsProcessing(false);
+
+        return;
+      }
+
+      patchExpense({
+        id: currentExpense.id,
+        user: formUser,
+        title: formTitle,
+        category: formCategory,
+        amount: formAmount,
+        date: formDate,
+        note: formNote,
+      })
+        .then(() => {
+          dispatch(setModalState({ variant: null }));
+          dispatch(fetchExpenses());
+        })
+        .catch(() => {
+          setIsProcessing(false);
+        });
+    }
+  }
+
+  const handleCancel = () => {
+    dispatch(setModalState({ variant: null }));
+  };
+
+  useEffect(() => {
+    if (!currentExpenseId) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    getExpenseById(currentExpenseId)
+      .then(result => setCurrentExpense(result))
+      .finally(() => setIsLoading(false));
+  }, [currentExpenseId]);
+
+  useEffect(() => {
+    if (!currentExpense) {
+      return;
+    }
+
+    const {
+      user,
+      title,
+      category,
+      amount,
+      date,
+      note,
+    } = currentExpense;
+
+    setFormUser(user);
+    setFormTitle(title);
+    setFormCategory(category);
+    setFormAmount(amount);
+    setFormDate(date);
+    setFormNote(note);
+  }, [currentExpense]);
+
+  return (
+    <>
+      {isLoading && <Loader className="mx-auto is-size-1" />}
+      {!isLoading && (
+        <form onSubmit={handleFormSubmit}>
+        <Form.Field>
+          <Form.Label>User</Form.Label>
+          <Form.Control>
+            <Form.Input
+              type="text"
+              value={formUser}
+              onChange={(e) => {
+                return setFormUser(e.target.value);
+              }}
+              required
+            />
+          </Form.Control>
+        </Form.Field>
+
+        <Form.Field>
+          <Form.Label>Title</Form.Label>
+          <Form.Control>
+            <Form.Input
+              type="text"
+              value={formTitle}
+              onChange={(e) => {
+                return setFormTitle(e.target.value);
+              }}
+              required
+            />
+          </Form.Control>
+        </Form.Field>
+
+        <Form.Field>
+          <Form.Label>Category</Form.Label>
+          <Form.Field>
+            <Form.Control>
+              <Form.Select
+                value={formCategory}
+                className="is-fullwidth"
+                onChange={(e) => {
+                  return setFormCategory(e.target.value);
+                }}
```

Comment:
why don't you write this way?
```suggestion
                onChange={(e) => setFormCategory(e.target.value) }
```